### ISSUE TITLE
IllegalArgumentException when s-match-examples started for the first time

### ISSUE THREAD

**Issue by Tuchkata** [s-match/s-match-examples#2]
Hi,

I am currently developing my master thesis and I need a Ontology mapping/matching algorithms and I've found s-match. I tried to build all needed artifacts and start the example, but when I run it, I get the following exception:
`Exception in thread "main" java.lang.IllegalArgumentException: the value should be from 0 to 4
    at it.unitn.disi.smatch.async.AsyncTask.setProgress(AsyncTask.java:289)
    at it.unitn.disi.smatch.async.AsyncTask.progress(AsyncTask.java:589)
    at it.unitn.disi.smatch.async.AsyncTask.progress(AsyncTask.java:573)
    at it.unitn.disi.smatch.preprocessors.RunnableDefaultContextPreprocessor.buildCLabs(RunnableDefaultContextPreprocessor.java:169)
    at it.unitn.disi.smatch.preprocessors.RunnableDefaultContextPreprocessor.preprocess(RunnableDefaultContextPreprocessor.java:134)
    at it.unitn.disi.smatch.MatchManager.preprocess(MatchManager.java:314)
    at it.unitn.disi.smatch.MatchManager.offline(MatchManager.java:281)
    at it.unitn.disi.smatch.examples.SMatchAPIDemo.example(SMatchAPIDemo.java:52)
    at it.unitn.disi.smatch.examples.SMatchAPIDemo.main(SMatchAPIDemo.java:21)`

Any ideas? I haven't changed anything in the default implementation.

**Comment by Tuchkata** [comment]
I've found the problem. When we first use the source with only 1 node, we set the property "total" in the preprocessor to 4 \* nodeCount = 4. The target tree has 5 nodes, which is with 1 more than the total and that is why the illegal argument has been thrown.
What I've done as a workaround is the following : I've changed the access modifier of the AsyncTask#setTotal(long) to public and I set it to 0 before executing the 
`mm.offline(t);`
in SMatchAPIDemo#example(). Doing this, the total is set independently for every single tree.
As this is just a workaround, I'll take a deeper look at the problem in order to reset the total count for every tree.
